[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66967
* #66452
* #67144
* __->__ #67143
* #66451
* #66450



Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

huiguoo committed:
[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"


Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

ZolotukhinM commented on torch/csrc/jit/tensorexpr/codegen.cpp:
```diff
@@ -80,6 +82,50 @@ void CodeGen::call_with_numel(void** args, int64_t numel) {
       false, "This codegen backend does not implement call_with_numel");
 }
+
+StmtPtr insertMemNodes(std::unordered_set<BufPtr>& interm_bufs, StmtPtr stmt) {
```
Nit: Let's be more precise with names. MemNodes can mean many different things, why do we need to change it from `insertAllocFree`?

ZolotukhinM reviewed (approved):
Looks good!

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).

huiguoo committed:
Update on "[tensorexpr] Move the allocation of intermediate buffers from TEK to CodeGen"




Differential Revision: [D31881151](https://our.internmc.facebook.com/intern/diff/D31881151)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881151).
